go/enclave/db/rawdb/accessors_chain.go

```diff
@@ -207,6 +207,25 @@ func ReadL2HeadRollup(kv ethdb.KeyValueReader, l1Head *common.L1BlockHash) (*com
 	return &l2Head, nil
 }
 
+// ReadCanonicalBatchHash retrieves the hash of the canonical batch at a given height.
+func ReadBatchBySequenceNum(db ethdb.Reader, number uint64) (*common.L2BatchHash, error) {
+	// Get it by hash from leveldb
+	data, err := db.Get(batchSeqHashKey(number))
+	if err != nil {
+		return nil, errutil.ErrNotFound
+	}
+	hash := gethcommon.BytesToHash(data)
+	return &hash, nil
+}
+
+// WriteCanonicalHash stores the hash assigned to a canonical batch number.
+func WriteBatchBySequenceNum(db ethdb.KeyValueWriter, l2Head *core.Batch) error {
+	if err := db.Put(batchSeqHashKey(l2Head.Header.SequencerOrderNo.Uint64()), l2Head.Hash().Bytes()); err != nil {
```

Comment:
I wonder if we should check that no batch already exists with this seq number? Like put if empty, err if already exists. Might be a good sanity check if it's an assumption we expect to hold true